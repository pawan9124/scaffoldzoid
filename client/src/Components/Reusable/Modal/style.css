.modal_body_position {
  margin: 10% 0 0 30%;
}
.modal_body {
  display: flex;
  flex-direction: row;
  justify-content: flex-end;
}
