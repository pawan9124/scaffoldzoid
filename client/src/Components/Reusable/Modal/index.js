import React, { useEffect } from "react";
import { makeStyles } from "@material-ui/core/styles";
import Modal from "@material-ui/core/Modal";
import Button from "../Button";
import "./style.css";

const useStyles = makeStyles((theme) => ({
  paper: {
    position: "absolute",
    width: 400,
    backgroundColor: theme.palette.background.paper,
    border: "2px solid #000",
    boxShadow: theme.shadows[5],
    padding: theme.spacing(2, 4, 3),
  },
}));

export default function SimpleModal(props) {
  const classes = useStyles();
  // getModalStyle is not a pure function, we roll the style only on the first render
  const [open, setOpen] = React.useState(false);

  useEffect(() => {
    setOpen(props.isOpen);
  }, [props.isOpen]);

  const handleClose = () => {
    setOpen(false);
    props.setIsOpen(false);
  };

  const body = (
    <div className={`${classes.paper} modal_body_position`}>
      <h2 id="simple-modal-title">
        {props.eventType === "add" ? "Add" : "Edit"} Orange Types
      </h2>
      <p id="simple-modal-description">
        {props.eventType === "add" ? "Add" : "Edit"} orange types to your
        inventory and provide various flavors to the user.
      </p>
      <div>{props.children}</div>
      <div className="modal_body">
        <Button
          type="submit"
          onClick={(e) => handleClose()}
          buttonClass="default__button profile_button"
          buttonLabel="Close"
        />
        <Button
          type="submit"
          onClick={(e) => props.handleOrangeEvent()}
          buttonClass="default__button profile_button"
          buttonLabel="Save"
        />
      </div>
    </div>
  );

  return (
    <div>
      <Modal
        open={open}
        onClose={handleClose}
        className="modal_position"
        aria-labelledby="simple-modal-title"
        aria-describedby="simple-modal-description"
      >
        {body}
      </Modal>
    </div>
  );
}
