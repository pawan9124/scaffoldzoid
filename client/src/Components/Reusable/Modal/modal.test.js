import React from "react";
import { shallow } from "enzyme";
import { findByTestAttr, checkProps } from "../../../testUtils";
import Modal from "./index";
import renderer from "react-test-renderer";

const props = {
  children: "<div>child</div>",
  handleOrangeEvent: jest.fn(),
  isOpen: true,
  setIsOpen: jest.fn(),
};

describe("Taking snap shot", () => {
  it("Taking the snapshot of the Component", () => {
    const tree = renderer.create(<Modal {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe("Checking the render of component", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<Modal {...props} />);
  });
  it("Should Render a component", () => {
    const button = findByTestAttr(wrapper, "ModalComponent");
    expect(button.length).toBe(1);
  });
});

describe("Checking PropTypes", () => {
  it("Should Not throw a warning", () => {
    const expectedProps = props;

    const propsError = checkProps(Modal, expectedProps);
    expect(propsError).toBeUndefined();
  });
});
