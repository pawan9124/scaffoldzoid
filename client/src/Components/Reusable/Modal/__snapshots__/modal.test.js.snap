// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Taking snap shot Taking the snapshot of the Component 1`] = `<div />`;
